akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterActorSetImpl.scala

```diff
@@ -0,0 +1,178 @@
+          // FIXME should we delay initial tick with random fraction of keepAliveInterval to avoid thundering herd?
+          timers.startTimerWithFixedDelay(Tick, settings.keepAliveInterval)
+          Behaviors.receiveMessage {
+            case Tick =>
+              identities.foreach(id => shardingRef ! ShardingEnvelope(id, EntityParent.Ping))
+              Behaviors.same
+          }
+        }
+      }
+  }
+
+}
+@InternalApi
+private[akka] class ClusterActorSetImpl(system: ActorSystem[_]) extends ClusterActorSet {
+
+  private val typeKeyCounter = new AtomicInteger(0)
+
+  override def init[T](numberOfEntities: Int, behaviorFactory: EntityId => Behavior[T], stopMessage: T): Unit =
+    init(ClusterActorSetSettings(system), numberOfEntities, behaviorFactory, stopMessage)
+
+  override def init[T](
+      settings: ClusterActorSetSettings,
+      numberOfEntities: Int,
+      behaviorFactory: EntityId => Behavior[T],
+      stopMessage: T): Unit = {
+    val identities = (0 to numberOfEntities).map(_.toString).toSet
+    init(settings, identities, behaviorFactory, stopMessage)
+  }
+
+  override def init[T](identities: Set[EntityId], behaviorFactory: EntityId => Behavior[T], stopMessage: T): Unit =
+    init(ClusterActorSetSettings(system), identities, behaviorFactory, stopMessage)
+
+  override def init[T](
+      settings: ClusterActorSetSettings,
+      identities: Set[EntityId],
+      behaviorFactory: EntityId => Behavior[T],
+      stopMessage: T): Unit = {
+    val setId = typeKeyCounter.incrementAndGet()
+    val entityTypeKey = EntityTypeKey[EntityParent.Command](s"cluster-actor-set-$setId")
+
+    // Since we know up front exactly what entity ids will exist and the number will be low 1:1 is fine as it will
+    // balance the set as good as possible across the cluster
+    val numberOfShards = identities.size
+
+    val shardingSettings = {
+      val settingsFromConfig =
+        ClusterShardingSettings.fromConfig(
+          // defaults in akka.cluster.sharding but allow overrides specifically for actor-set
+          system.settings.config.getConfig("akka.cluster.actor-set.sharding"))
+
+      new ClusterShardingSettings(
+        numberOfShards,
+        settingsFromConfig.role,
+        settingsFromConfig.dataCenter,
+        false, // remember entities disabled
+        "",
+        "",
+        Duration.Zero, // passivation disabled
+        settingsFromConfig.shardRegionQueryTimeout,
+        StateStoreModeDData,
+        settingsFromConfig.tuningParameters,
+        settingsFromConfig.coordinatorSingletonSettings)
+    }
+
+    val entity =
+      Entity(entityTypeKey)(ctx => EntityParent(ctx, behaviorFactory, stopMessage))
+        .withSettings(shardingSettings)
+        .withStopMessage(EntityParent.Stop)
+
+    val shardingRef = ClusterSharding(system).init(entity)
+
+    system.systemActorOf(KeepAlivePinger(settings, identities, shardingRef), s"clusterActorSetPinger-$setId")
```

Comment:
would also be nice if the KeepAlivePinger could be used standalone because then the actors can be initialized as ordinary Sharded actors that can receive ordinary messages, but are also kept-alive.

Reply:
but if we keep the parent this comment is obsolete